sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/ServerCertificateCheck.kt

```diff
@@ -19,44 +19,107 @@
  */
 package org.sonarsource.kotlin.checks
 
-import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtCatchClause
 import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtThrowExpression
+import org.jetbrains.kotlin.psi.KtVisitorVoid
 import org.jetbrains.kotlin.resolve.BindingContext
-import org.jetbrains.kotlin.resolve.BindingContext.CLASS
-import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
-import org.jetbrains.kotlin.resolve.descriptorUtil.getAllSuperClassifiers
 import org.sonar.check.Rule
 import org.sonarsource.kotlin.api.AbstractCheck
+import org.sonarsource.kotlin.api.ConstructorMatcher
+import org.sonarsource.kotlin.api.FunMatcher
 import org.sonarsource.kotlin.plugin.KotlinFileContext
 
+
+private const val CERTIFICATE_EXCEPTION = "CertificateException"
+
 @Rule(key = "S4830")
 class ServerCertificateCheck : AbstractCheck() {
     companion object {
-        private val methodNames = setOf("checkClientTrusted", "checkServerTrusted")
-        private val firstArgRegex = Regex("""Array<(out )?X509Certificate\??>\??""")
-        private val secondArgRegex = Regex("""String\??""")
+        val funMatchers = listOf(
+            FunMatcher {
+                definingSupertype = "javax.net.ssl.X509TrustManager"
+                withNames("checkClientTrusted", "checkServerTrusted")
+            },
+            FunMatcher {
+                definingSupertype = "javax.net.ssl.X509ExtendedTrustManager"
+                withNames("checkClientTrusted", "checkServerTrusted")
+            })
     }
```

Comment:
Another artifact from this being an older rule: We have actually stopped declaring 'static' values such as this one inside of the companion object. Instead, we usually declare them above the class, where you also correctly declared the const for `CertificateException`. Since we are touching this, we might as well get rid of the companion object here.

Reply:
I think part of the reason was also the fact that companion objects are not really very natural to the language and overall just a bit clunky. Let us discuss this elsewhere in more detail, though, if it interests you!